### PR TITLE
Added RepVGG models

### PR DESCRIPTION
**Pull request by Anna-Mironova**

### CODE REVIEW

**Comment by eaidova** [comment]
jenkins please retry a build

**Review by Wovchena — APPROVED** [review]

**Comment by Wovchena** [comment]
jenkins please retry a build

**Comment by Anna-Mironova** [comment]
jenkins please retry a build

File: models/public/repvgg-a0/pre-convert.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('input_dir', type=Path)
+    parser.add_argument('output_dir', type=Path)
+    args = parser.parse_args()
+
+    subprocess.run([sys.executable, '--',
+        str(args.input_dir / 'convert.py'),
```
[the remark this answers is not in the document]
Comment (Anna-Mironova) [reply models/public/repvgg-a0/pre-convert.py]: Updated. Removed the use of external script

**Review by eaidova — APPROVED** [review]

File: models/public/repvgg-a0/pre-convert.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    parser.add_argument('output_dir', type=Path)
+    args = parser.parse_args()
+
+    sys.path.append(str(args.input_dir))
+    repvgg = importlib.import_module('repvgg')
+
+    train_model = repvgg.create_RepVGG_A0(deploy=False)
+
+    checkpoint = load(str(args.input_dir / 'RepVGG-A0-train.pth'))
+    if 'state_dict' in checkpoint:
```
Comment (IRDonch) [inline models/public/repvgg-a0/pre-convert.py:37]: Shouldn't it be known ahead of time whether this is true? The input checkpoint file is fixed.

**Review by vladimir-dudnik — COMMENTED** [review]
Is there are any special reasons to add four mostly similar (according to the description) models? Will not it be enough to add only either B0 or B1 variant?

**Comment by eaidova** [comment]
> Is there are any special reasons to add four mostly similar (according to the description) models? Will not it be enough to add only either B0 or B1 variant?

Different complexity and different accuracy (b3 is SoTA for ImageNet right now), we selected 4 most significant from 12

**Comment by vladimir-dudnik** [comment]
might be limit to B0 and B3 then?

File: models/public/repvgg-a0/pre-convert.py (hunk 2 above)
Comment (Anna-Mironova) [reply models/public/repvgg-a0/pre-convert.py]: Сondition removed

**Comment by eaidova** [comment]
Jenkins please retry a build

**Review by IRDonch — APPROVED** [review]

**Review by vladimir-dudnik — CHANGES_REQUESTED** [review]
could you please add models to test cases?

**Comment by Anna-Mironova** [comment]
> could you please add models to test cases?

Done

**Review by vladimir-dudnik — APPROVED** [review]

**Comment by eaidova** [comment]
Jenkins please retry a build
